jena-jdbc/jena-jdbc-core/pom.xml

```diff
@@ -39,17 +39,28 @@
 			<version>3.10.0-SNAPSHOT</version>
 		</dependency>
 
-    <!-- Jena Driver configures Log4j directly -->
-    <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-log4j12</artifactId>
-    </dependency>
+		<!-- Jena Driver configures Log4j directly -->
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-log4j12</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>log4j</groupId>
+			<artifactId>log4j</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
+			<artifactId>log4j-api</artifactId>
+		</dependency>
+
+		<!-- Intercept direct use of log4j2 -->
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
```

Comment:
I think this is unnecessary, the only Log4j direct usage is to configure Log4j, the code itself always uses SLF4J for its logging

Reply:
OK so we agree on this however the code doesn't build due to the missing dependency. I wasn't able to build without the additions as proposed.

Reply:
I'm unclear here. I tried with JDK10 and had to add the log4j1 dependency but not the log4j2 dependencies.

In the code I can only find imports of `org.apache.log4j`, which is log4j1, not `org.apache.logging.log4j` which is log4j2.

Elasticsearch does use log4j2 and Jena adds adapters theer for that.

Reply:
More:

Confirmed I have only have to add a single log4j1 dependency.
The error is otherwise from the maven-javadoc-plugin, not the compiler:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:jar (attach-javadocs) on project jena-jdbc-driver-bundle: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - /home/afs/Jena/jena-jdbc/jena-jdbc-core/src/main/java/org/apache/jena/jdbc/JenaDriver.java:42: error: package org.apache.log4j does not exist
[ERROR] import org.apache.log4j.PropertyConfigurator;
[ERROR]                        ^
[ERROR] 
[ERROR] Command line was: /usr/lib/jvm/default-java/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/home/afs/Jena/jena-jdbc/jena-jdbc-driver-bundle/target/apidocs' dir.
[ERROR] 
```

Reply:
Agreed. This is the behavior I get as well. 
My thinking was that the dependency needed to be available at compile time as oppose to only runtime... but maybe this thinking is wrong. 
The generated Javadoc look fine but the error prevents a stable build.

Reply:
From what I understand, the offending class [org.apache.log4j.PropertyConfigurator](https://logging.apache.org/log4j/2.0/log4j-1.2-api/apidocs/index.html?org/apache/log4j/PropertyConfigurator.html) resides in log4j 2.0... the runtime dependencies are detailed [as follows](https://logging.apache.org/log4j/2.0/runtime-dependencies.html) with the [Maven Artifacts detailed here](https://logging.apache.org/log4j/2.0/maven-artifacts.html).

Reply:
This code predates Log4j 2.x so shouldn't have any dependency upon it

Reply:
Wrong PropertyConfigurator.java  - there are two, one in log4j1 and one in log4j2 for compatibility code `log4j-1.2-api/src/main/java/org/apache/log4j/PropertyConfigurator.java`.

Jena is picking up the one from log4j1. So the log4j2 dependencies aren't need but the log4j1 is. Either will do to compile but log4j1 is intended.